### PR TITLE
feat: enable request bodies decompression

### PR DESCRIPTION
**Pull request by etolbakov**
### Description

enable request bodies decompression

### How was this PR tested?

this setting hasn't been tested

### Related issue
https://github.com/quickwit-oss/quickwit/issues/3990

### CODE REVIEW

File: quickwit/quickwit-serve/src/rest.rs [hunk 1]
```diff
@@ -152,6 +153,7 @@ pub(crate) async fn start_rest_server(
                 .compress_when(compression_predicate),
         )
         .layer(cors)
+        .layer(DecompressionLayer::new().gzip(true))
```
Comment (guilload) [inline quickwit/quickwit-serve/src/rest.rs:156]: Can we support `zstd` as well?

**Comment by guilload** [comment]
Would you be able to test this? You could follow our getting started tuto and tweak the cURL command:
https://quickwit.io/docs/main-branch/get-started/quickstart.

**Comment by etolbakov** [comment]
Currently  `warp` (web framework) does not handle compressed bodies out of the box.
https://github.com/seanmonstar/warp/issues/1035

**Comment by guilload** [comment]
Yes, my bad. I assumed that `warp` is compatible with most `tower-http` layers, but that's no the case. Thanks for doing the discovery work.

**Comment by guilload** [comment]
@unpervertedkid, the best way to go about this is adding support in `warp`. I've looked into it, it's not too hard.

**Pull request closed without merging**